### PR TITLE
New dynamo tables

### PR DESCRIPTION
**Pull request by karkir0003**
# Trainspace DB Table

**What user problem are we solving?**
DDB table for trainspaces
**What solution does this PR provide?**
DDB table
**Testing Methodology**
used ipython + main method
**Any other considerations**
make sure it works

### CODE REVIEW

**Comment by karkir0003** [comment]
@farisdurrani @dwu359 thoughts?

File: backend/aws_helpers/dynamo_db_utils/trainspace.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            trainspace_id, status=status, modified=get_current_timestamp()
+        )
+        return '{"status": "success"}'
+    except:
+        entryData["status"] = status
+        createTrainspaceData(entryData)
+        return '{"status": "success"}'
+
+
+def validate_keys(requestData: dict, required_keys: list[str]) -> bool:
```
Comment (karkir0003) [inline backend/aws_helpers/dynamo_db_utils/trainspace.py:222]: @farisdurrani can you update the pydocs as well for the trainspace table

**Comment by karkir0003** [comment]
@farisdurrani found the problem with the build check. the problem was in `test_send_email.py` and the attachment file path was invalid due to nextJS upgrade. I adjusted the test attachment array accordingly

**Review by farisdurrani — APPROVED** [review]

**Comment by farisdurrani** [comment]
@karkir0003 fixed the pydocs, see if it's okay

**Comment by farisdurrani** [comment]
Tested and worked

File: backend/aws_helpers/dynamo_db_utils/trainspace.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    "step",
+    "status",
+    "modified",
+    "train_model",
+]
+
+
+@dataclass
+class TrainspaceData(BaseData):
+    """Data class to hold the attribute values of a record of the execution-table DynamoDB table"""
```
Comment (karkir0003) [inline backend/aws_helpers/dynamo_db_utils/trainspace.py:33]: you mean trainspace table? @farisdurrani

**Review by karkir0003 — COMMENTED** [review]
tiny comment

**Comment by karkir0003** [comment]
@farisdurrani

File: backend/aws_helpers/dynamo_db_utils/trainspace.py (hunk 2 above)
Comment (farisdurrani) [reply backend/aws_helpers/dynamo_db_utils/trainspace.py:33]: Nooo, TrainspaceData. That is the convention we're using in our code

**Pull request merged**